test/gcs_taskrun_test.go

```diff
@@ -0,0 +1,184 @@
+// +build e2e
+
+/*
+Copyright 2018 Knative Authors LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
+	tb "github.com/knative/build-pipeline/test/builder"
+	knativetest "github.com/knative/pkg/test"
+	"github.com/knative/pkg/test/logging"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TestStorageTaskRun is an integration test that will verify GCS input resource runtime contract
+// - adds volumes with expected name
+// - places files in expected place
+
+func TestStorageTaskRun(t *testing.T) {
+	logger := logging.GetContextLogger(t.Name())
+
+	if _, checkgcloudErr := exec.LookPath("gcloud"); checkgcloudErr != nil {
+		logger.Infof("gcloud is not on path. So skipped test")
+		t.Skip()
+	}
+
+	projectCmd, err := exec.Command("gcloud", "config", "list", "project", "--format", "json").CombinedOutput()
+	if err != nil {
+		t.Fatalf("gcloud project config err %s", err)
+	}
+	project := GCPProject{}
+	if jsonErr := json.Unmarshal(projectCmd, &project); jsonErr != nil {
+		logger.Infof("gcloud project unmarshall output err %s. Skipping test", jsonErr)
+		t.Skip()
+	}
+
+	if project.GCPconfig.Name == "" {
+		logger.Infof("gcloud project configuration is empty. Skipping test")
+		t.Skip()
+	}
+
+	testSAName := AppendRandomString("e2etestsa")
+	// gcloud iam service-accounts create test
+	if saCmd, err := exec.Command("gcloud", "iam", "service-accounts", "create", testSAName).CombinedOutput(); err != nil {
+		logger.Infof("gcloud service account create output name %s: %s ; error: %s", testSAName, saCmd, err)
+		t.Skip()
+	}
+
+	c, namespace := setup(t, logger)
+
+	knativetest.CleanupOnInterrupt(func() { tearDown(t, logger, c, namespace) }, logger)
+	defer tearDown(t, logger, c, namespace)
+
+	authSec, saName := createGCSSecret(t, logger, namespace, project.GCPconfig.Name, testSAName)
+
+	defer func(saName string) {
+		if _, err := exec.Command("gcloud", "iam", "service-accounts", "delete", saName, "--quiet").CombinedOutput(); err != nil {
+			logger.Infof("Failed to delete gcloud service account %s with error %s", saName, err)
+		}
+	}(saName)
```

Comment:
just a note that if the test is interrupted, i dont think this will be executed, so you might want to use something like `knativetest.CleanupOnInterrupt` as well